pkg/applicationserver/applicationserver.go

```diff
@@ -105,7 +108,12 @@ var errListenFrontend = errors.DefineFailedPrecondition("listen_frontend", "fail
 
 // New returns new *ApplicationServer.
 func New(c *component.Component, conf *Config) (as *ApplicationServer, err error) {
-	ctx := log.NewContextWithField(c.Context(), "namespace", "applicationserver")
+	c.AddContextFiller(func(ctx context.Context) context.Context {
+		return tracer.NewContextWithTracer(ctx, packageName)
```

Comment:
The context filling is component-wide. The component `c` can be shared with multiple roles; the single binary `ttn-lw-stack start all` uses one component. If you add context fillers, they will be used by all other roles. So the request context of `NetworkServer` will be filled with the tracers of all other roles in the process.

So if you need per-role context filling for the request context, you need to register a hook. See how `c.GRPC.Register{Unary,Stream}Hook()` is being used.